packages/devtools_app/lib/src/screens/performance/panes/timeline_events/timeline_events_controller.dart

```diff
@@ -375,46 +378,34 @@ class TimelineEventsController extends PerformanceFeatureController
       hasProcessedTimelineEventsForFrame =
           perfettoController.processor.hasProcessedEventsForFrame(frame.id);
       if (!hasProcessedTimelineEventsForFrame) {
+        final timelineEventsUnavailable =
```

Comment:
Why do we do this twice? Is it worth putting into a helper?

Reply:
added comments in the code to explain. and added a helper.